Sources/PostgreSQL/Database.swift

```diff
@@ -38,7 +40,47 @@ public final class Database: ConnInfoInitializable {
 
         return try connection.execute(query, values)
     }
+	
+	public func listen(to channel: String, callback: @escaping (Notification) -> Void) throws {
+		try background {
+			do {
+				let connection = try self.makeConnection()
+				
+				try self.execute("LISTEN \(channel)", on: connection)
+				
+				while true {
+					if connection.connected != true {
+						throw DatabaseError.cannotEstablishConnection(connection.error)
+					}
+					
+					PQconsumeInput(connection.connection)
+					
+					while let pgNotify = PQnotifies(connection.connection) {
+						let notification = Notification(relname: pgNotify.pointee.relname, extra: pgNotify.pointee.extra, be_pid: pgNotify.pointee.be_pid)
+						
+						callback(notification)
+						
+						PQfreemem(pgNotify)
+					}
+				}
+			}
+			catch {
+				fatalError("\(error)")
+			}
+		}
+	}
 
+	public func notify(channel: String, payload: String?, on connection: Connection? = nil) throws {
+		let connection = try connection ?? makeConnection()
+		
+		if let payload = payload {
+			try execute("NOTIFY \(channel), '\(payload)'", on: connection)
+		}
+		else {
+			try execute("NOTIFY \(channel)", on: connection)
+		}
+	}
+	
     public func makeConnection() throws -> Connection {
         return try Connection(conninfo: conninfo)
     }
```

Comment:
I recommend moving the makeConnection function above the listen and notify functions. It will keep the flow of the document consistent.